lms/services/canvas_api.py

```diff
@@ -50,6 +51,22 @@ def get_token(self, authorization_code):
             response.parsed_params.get("expires_in"),
         )
 
+    def get_refreshed_token(self, refresh_token):
+        response = self._helper.validated_response(
+            self._helper.refresh_token_request(refresh_token),
+            CanvasRefreshTokenResponseSchema,
+        )
+
+        new_access_token = response.parsed_params["access_token"]
+
+        self._save(
+            new_access_token,
+            response.parsed_params.get("refresh_token", refresh_token),
```

Comment:
If no new refresh token is given in the refresh token response, then we keep the original refresh token in the DB with the new access token. This is the case with Canvas: they want you to keep the same refresh token and keep on using it